lego/apps/polls/serializers.py

```diff
@@ -0,0 +1,150 @@
+
+    comments = CommentSerializer(read_only=True, many=True)
+    comment_target = CharField(read_only=True)
+
+    options = OptionSerializer(many=True)
+    total_votes = IntegerField(read_only=True)
+
+    has_answered = serializers.SerializerMethodField()
+
+    def get_has_answered(self, obj):
+        user = self.context["request"].user
+        return obj.get_has_answered(user)
+
+    class Meta:
+        model = Poll
+        fields = (
+            "id",
+            "created_at",
+            "valid_until",
+            "title",
+            "description",
+            "options",
+            "total_votes",
+            "tags",
+            "comments",
+            "comment_target",
+            "has_answered",
+            "pinned",
+        )
+
+
+class DetailedPollSerializer(TagSerializerMixin, BasisModelSerializer):
+
+    comments = CommentSerializer(read_only=True, many=True)
+    comment_target = CharField(read_only=True)
+
+    options = OptionSerializer(many=True)
+    total_votes = IntegerField(read_only=True)
+
+    has_answered = serializers.SerializerMethodField()
+
+    def get_has_answered(self, obj):
+        user = self.context["request"].user
+        return obj.get_has_answered(user)
+
+    class Meta:
+        model = Poll
+        fields = (
+            "id",
+            "created_at",
+            "valid_until",
+            "title",
+            "description",
+            "options",
+            "total_votes",
+            "comments",
+            "comment_target",
+            "tags",
+            "has_answered",
+            "pinned",
+        )
+
+
+class PollCreateSerializer(TagSerializerMixin, BasisModelSerializer):
+
+    options = OptionSerializer(many=True)
+
+    class Meta:
+        model = Poll
+        fields = (
+            "id",
+            "created_at",
+            "title",
+            "description",
+            "options",
+            "total_votes",
+            "tags",
+            "pinned",
+            "valid_until",
+        )
+
+    @transaction.atomic
+    def create(self, validated_data):
+        options_data = validated_data.pop("options")
+        poll = super().create(validated_data)
+        for option in options_data:
+            Option.objects.create(poll=poll, **option)
+        return poll
+
+
+class PollUpdateSerializer(TagSerializerMixin, BasisModelSerializer):
+
+    options = OptionUpdateSerializer(many=True)
+
+    class Meta:
+        model = Poll
+        fields = (
+            "id",
+            "created_at",
+            "title",
+            "description",
+            "options",
+            "total_votes",
+            "tags",
+            "pinned",
+        )
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        options = validated_data.pop("options")
+
+        # Update the regular fields that aren't options first
+        super().update(instance, validated_data)
+
+        # Delete options that aren't in the received list
```

Comment:
I think you should be able to remove an option if there has been a mistake. The votes will be lost, but I think that's fine. However, since there is no relation between the user and an option, the user will still be listed as answered. Which maybe isn't the best behavior, but there's no way to remove the user along with the vote. Thoughts?